rust/zypp-agama/zypp-agama-sys/c-layer/callbacks.cxx

```diff
@@ -118,6 +119,163 @@ struct DownloadProgressReceive : public zypp::callback::ReceiveReport<
 
 static DownloadProgressReceive download_progress_receive;
 
+struct DownloadResolvableReport : public zypp::callback::ReceiveReport<
+                                      zypp::repo::DownloadResolvableReport> {
+  struct DownloadResolvableCallbacks *callbacks;
+
+  DownloadResolvableReport() { callbacks = NULL; }
+
+  void set_callbacks(DownloadResolvableCallbacks *callbacks_) {
+    callbacks = callbacks_;
+  }
+
+  virtual Action problem(zypp::Resolvable::constPtr resolvable_ptr, Error error,
+                         const std::string &description) {
+    if (callbacks != NULL && callbacks->problem != NULL) {
+      enum DownloadResolvableError error_enum;
+      switch (error) {
```

Comment:
NP: IMHO we should factor out these boilerplate enum conversions to separate short methods to make the rest more readable